spirv_glsl.cpp

```diff
@@ -7585,7 +7618,7 @@ bool CompilerGLSL::remove_unity_swizzle(uint32_t base, string &op)
 	auto &type = expression_type(base);
 
 	// Sanity checking ...
-	assert(type.columns == 1 && type.array.empty());
+	assert(type.columns == 1); //  && type.array.empty()
```

Comment:
Why remove array check?

Reply:
Looks like the same case as with the removal of `ir.reset_all_of_type<SPIRExpression>();` to me. Updated shaders work with this change reverted.